explainers/hidden-content-explainer.md

```diff
@@ -12,28 +12,21 @@ This is an explainer for two closely related features:
         hidden from the user (similar to `content-visibility: hidden`), but the
         content remains searchable via user-agent find-in-page algorithms.
 
-2. The `beforematch` event allows developers to display content to the user in
-    response to actions which scroll the page to a target element. It is fired
-    at render timing just before these actions scroll the page. The applicable
-    actions are the following:
-    * There is a new find-in-page (ctrl+f) active match, in which case the event
-      is fired with the matched element as the target.
-    * There is an element fragment navigation (`example.com/#foo`) or the fragment
-      is changed by script (`window.location.hash = ...`), in which case the event
-      is fired with the matched element as the target.
+2. The `beforematch` event allows developers to display `hidden-matchable`
+    content to the user in response to searches which scroll the page to some
+    target text. The event is fired at render timing on the nearest ancestor
+    block-level element for these cases:
+    * There is a new find-in-page (ctrl+f)
+      [active match](https://html.spec.whatwg.org/multipage/interaction.html#fip-matches)
+      which is located inside an element with `content-visibility:
+      hidden-matchable` style.
     * There is a [scroll-to-text](https://github.com/WICG/ScrollToTextFragment)
-      navigation (`example.com/#:~:text=foo`), in which case the event is fired with
-      the matched element as the target.
-
-Note that the 'matched element' in this document refers to one of the following:
-* In the find-in-page and scroll-to-text cases: The closest block-level ancestor
-  element in the flat tree containing the matching text.
-  If the match spans multiple block-level elements, then the first block-level
-  element is used.
-  Since the flat tree is used to determine the element to fire the event on,
-  shadow boundaries have no impact.
-* In the element fragment case: The element which was selected by id from the
-  fragment.
+      navigation (`example.com/#:~:text=foo`), where the target text is located
+      inside a `content-visibility: hidden-matchable` element.
+
+If the matching text spans multiple block-level elements, then the first
```

Comment:
We need to decide whether we fire an event on the nearest ancestor or nearest ancestor with content-visibility: hidden-matchable (and make that clear)

Reply:
hmmm now that we are only firing on hidden-matchable, i suppose it could make sense to only fire on that element instead of going to the block-level element and bubbling it from there. what do you think?
Or are you just saying for the case where it spans multiple block level elements?

Reply:
Either way, I think we should find the nearest content-visibility: hidden-matchable element and fire it on that element (with bubbling). Same caveats apply: if the text spans multiple block level elements all of which have hidden-matchable, then just do it on the first one. I think it would be a rare case though, and I'm not sure we can find text across layout containment which hidden-matchable would put in.

Reply:
Yeah that's fair if we are only firing it on hidden-matchable content now.
I updated this and the previous text to say that we will only fire it on hidden-matchable content.